app/src/test/java/com/kickstarter/viewmodels/CreatorDashboardReferrerBreakdownHolderViewModelTest.java

```diff
@@ -0,0 +1,369 @@
+    this.vm.outputs.internalReferrerPercentText().subscribe(this.internalReferrerPercentText);
+    this.vm.outputs.pledgedViaCustomLayoutIsGone().subscribe(this.pledgedViaCustomLayoutIsGone);
+    this.vm.outputs.pledgedViaExternalLayoutIsGone().subscribe(this.pledgedViaExternalLayoutIsGone);
+    this.vm.outputs.pledgedViaInternalLayoutIsGone().subscribe(this.pledgedViaInternalLayoutIsGone);
+    this.vm.outputs.projectAndAveragePledge().subscribe(this.projectAndAveragePledge);
+    this.vm.outputs.projectAndCustomReferrerPledgedAmount().subscribe(this.projectAndCustomReferrerPledgedAmount);
+    this.vm.outputs.projectAndExternalReferrerPledgedAmount().subscribe(this.projectAndExternalReferrerPledgedAmount);
+    this.vm.outputs.projectAndInternalReferrerPledgedAmount().subscribe(this.projectAndInternalReferrerPledgedAmount);
+  }
+
+  @Test
+  public void testCustomReferrerColor() {
+    final Project project = ProjectFactory.project();
+    final ProjectStatsEnvelope statsEnvelope = ProjectStatsEnvelopeFactory.projectStatsEnvelope();
+
+    setUpEnvironment(environment());
+    this.vm.inputs.projectAndStatsInput(Pair.create(project, statsEnvelope));
+    this.customReferrerColor.assertValues(ReferrerColor.CUSTOM.getReferrerColor());
+  }
+
+  @Test
+  public void testCustomReferrerPercent() {
+    final Project project = ProjectFactory.project();
+    final ProjectStatsEnvelope.ReferrerStats fiftyFivePercentCustomReferrer = ProjectStatsEnvelopeFactory.ReferrerStatsFactory.referrerStats().toBuilder()
+      .percentageOfDollars(55f)
+      .referrerType(ReferrerType.CUSTOM.getReferrerType())
+      .build();
+    final List<ProjectStatsEnvelope.ReferrerStats> referrerList = Arrays.asList(
+      fiftyFivePercentCustomReferrer);
+    final ProjectStatsEnvelope statsEnvelope = ProjectStatsEnvelopeFactory.projectStatsEnvelope().toBuilder().referralDistribution(referrerList).build();
+
+    setUpEnvironment(environment());
+    this.vm.inputs.projectAndStatsInput(Pair.create(project, statsEnvelope));
+    this.customReferrerPercent.assertValues(55f);
+  }
+
+  @Test
+  public void testCustomReferrerPercentText() {
+    final Project project = ProjectFactory.project();
+    final ProjectStatsEnvelope.ReferrerStats fiftyFivePercentCustomReferrer = ProjectStatsEnvelopeFactory.ReferrerStatsFactory.referrerStats().toBuilder()
+      .percentageOfDollars(55f)
+      .referrerType(ReferrerType.CUSTOM.getReferrerType())
+      .build();
+    final List<ProjectStatsEnvelope.ReferrerStats> referrerList = Arrays.asList(
+      fiftyFivePercentCustomReferrer);
+    final ProjectStatsEnvelope statsEnvelope = ProjectStatsEnvelopeFactory.projectStatsEnvelope().toBuilder().referralDistribution(referrerList).build();
+
+    setUpEnvironment(environment());
+    this.vm.inputs.projectAndStatsInput(Pair.create(project, statsEnvelope));
+    this.customReferrerPercentText.assertValues(NumberUtils.flooredPercentage(55f * 100f));
```

Comment:
if we're just asserting the percentage value here  (perecentText) maybe we could just assert "55%" since we don't have NumberUtils tests